src/endpoints/resources.js

```diff
@@ -10,71 +10,99 @@ const apiBaseUrl = process.env.API_BASE_URL
 resourcesRouter.post('/:resourceId', (req, res, next) => {
   console.log(`Received post to ${req.params.resourceId}`)
 
-  const resource = req.body
-  const resourceUri = resourceUriFor(req)
-  const timestamp = new Date()
-
-  const saveResource = resourceForSave(resource, req.params.resourceId, resourceUri, timestamp)
+  if (req.body.data === null) return next(new createError.BadRequest())
```

Comment:
Can all of this validation be moved into a method to avoid repeating and cluttering this method?

Reply:
good suggestion;  done.